core/src/main/java/hudson/FilePath.java

```diff
@@ -189,6 +193,11 @@
  */
 public final class FilePath implements Serializable {
     /**
+     * Maximum http redirects we will follow. This defaults to the same number as Firefox/Chrome tolerates.
+     */
+    private static final int MAX_REDIRECTS = 20;
```

Comment:
I thought that would be overkill since this is the same default as Firefox/Chrome which should cover most cases.